### PR TITLE
Fix conandata_yaml_linter

### PR DESCRIPTION
**Pull request by ericLemanissier**
---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
This fixes this error: https://github.com/conan-io/conan-center-index/actions/runs/3315312044/jobs/5475743773#step:9:411
It looks like it tests all files now: https://github.com/conan-io/conan-center-index/actions/runs/3330066012/jobs/5508069967#step:9:4802

**Comment by prince-chrismc** [comment]
https://github.com/conan-io/conan-center-index/actions/runs/3330066012/jobs/5508069967#step:9:360

Good catch no more "unbound" exception

File: linter/conandata_yaml_linter.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def main():
             f"title=config.yml schema error"
```
Comment (prince-chrismc) [inline linter/conandata_yaml_linter.py:73]:
```suggestion
            f"title=conandata.yml schema error"
```

**Review by prince-chrismc — COMMENTED** [review]
Might be worth changing line 65-67 https://github.com/conan-io/conan-center-index/pull/13779/files#diff-3de2595c4df5ac161e2b20c6423e6d37b439454ae55d1efff0118ecd138c0f1eR65

So its more clear looking at the job summary

```py
                            f"title=conandata.yml schema warning"
                            "::'patch_type' should have 'patch_source' as per https://github.com/conan-io/conan-center-index/blob/master/docs/conandata_yml_format.md#patches-fields"
                            " it is expected to have a source (e.g. a URL) to where it originates from to help with reviewing and consumers to evaluate patches\n"
```

**Review by prince-chrismc — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**